### PR TITLE
refactor: remove remaining unused code from cpp-subprocess

### PR DESCRIPTION
**Pull request by theStack**
This PR removes remaining code that is unused within the cpp-subprocess module (templates and commented out code). Happy to add more removals if anyone finds more unused parts. Note that there are some API functions of the `Popen` class that we don't use, e.g. `wait()`, `pid()`, `poll()`, `kill()`, but they sound IMHO common enough to be useful in the future, so not sure how deep we should go there.

### CODE REVIEW

**Comment by fanquake** [comment]
> Note that there are some API functions of the Popen class that we don't use, e.g. wait(), pid(), poll(), kill(), but they sound IMHO common enough to be useful in the future, so not sure how deep we should go there.

Deleting as much as possible seems fine. If code turns out to be needed in future (although I wouldn't think we are going to expand this interface), it can be retrieved from the git history.

**Comment by Sjors** [comment]
As long as #29868 works on top of this, it's fine by me. I'll do some testing.

**Review by hebasto — APPROVED** [review]
ACK 908c51fe4afeba0af500c6275027b1afa1b3bd19. It is compatible with ~https://github.com/bitcoin/bitcoin/pull/29961~  #29868.

I'd suggest to remove the mentioning of the `check_output` function from all comments. That function was not even pulled from the upstream.

**Comment by hebasto** [comment]
> > Note that there are some API functions of the Popen class that we don't use, e.g. wait(), pid(), poll(), kill(), but they sound IMHO common enough to be useful in the future, so not sure how deep we should go there.
> 
> Deleting as much as possible seems fine. If code turns out to be needed in future (although I wouldn't think we are going to expand this interface), it can be retrieved from the git history.

I agree.

File: src/util/subprocess.h [hunk 1]
```diff
@@ -904,8 +904,6 @@ class Streams
  * 3. wait()             - Wait for the child to exit.
  * 4. retcode()          - The return code of the exited child.
  * 5. pid()              - PID of the spawned child.
- * 6. poll()             - Check the status of the running child.
- * 7. kill(sig_num)      - Kill the child. SIGTERM used by default.
  * 8. send(...)          - Send input to the input channel of the child.
```
Comment (fjahr) [inline src/util/subprocess.h]: The numbering of this list is off after this change. That might be confusing to someone in the future. I would suggest either updating the following numbering or something like.

```
 * 6. (deleted)
 * 7. (deleted)
```

File: src/util/subprocess.h [hunk 2]
```diff
@@ -1009,27 +955,12 @@ class Popen
...
-  }
-*/
-
   int pid() const noexcept { return child_pid_; }
 
   int retcode() const noexcept { return retcode_; }
 
   int wait() noexcept(false);
 
-  int poll() noexcept(false);
```
Comment (hebasto) [inline src/util/subprocess.h:1027]: ed39d261dc056e700b79da769e085c8af389451d:

The `Popen::poll()` function is used in #29868. I hope that an improved/fixed Windows implementation of the `Popen::retcode()`, which I don't have now, will allow to drop it.

File: src/util/subprocess.h (hunk 1 above)
Comment (fjahr) [reply src/util/subprocess.h]: Ok, strange, I guess I was looking at an outdated version of the code or something.

**Comment by fjahr** [comment]
> ACK [908c51f](https://github.com/bitcoin/bitcoin/commit/908c51fe4afeba0af500c6275027b1afa1b3bd19). It is compatible with #29961.

@hebasto #29961 is this PR here, did you mean #29868 there?

**Comment by fjahr** [comment]
Code review ACK 8b52e7f628304e83b0e36fd97e617de0f71c5a62

**Review by hebasto — APPROVED** [review]
ACK 8b52e7f628304e83b0e36fd97e617de0f71c5a62.

**Comment by achow101** [comment]
ACK 8b52e7f628304e83b0e36fd97e617de0f71c5a62

**Pull request merged**